src/main/java/org/opentripplanner/middleware/controllers/api/TripHistoryController.java

```diff
@@ -0,0 +1,146 @@
+package org.opentripplanner.middleware.controllers.api;
+
+import com.mongodb.client.model.Filters;
+import org.bson.conversions.Bson;
+import org.eclipse.jetty.http.HttpStatus;
+import org.opentripplanner.middleware.auth.Auth0Connection;
+import org.opentripplanner.middleware.auth.Auth0UserProfile;
+import org.opentripplanner.middleware.models.TripRequest;
+import org.opentripplanner.middleware.models.OtpUser;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.persistence.TypedPersistence;
+import org.opentripplanner.middleware.utils.DateUtils;
+import org.opentripplanner.middleware.utils.HttpUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeParseException;
+import java.util.Date;
+
+import static com.mongodb.client.model.Filters.*;
+import static org.opentripplanner.middleware.utils.JsonUtils.logMessageAndHalt;
+
+/**
+ * Responsible for processing trip history related requests provided by MOD UI.
+ * To provide a response to the calling MOD UI in JSON based on the passed in parameters.
+ */
+public class TripHistoryController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TripHistoryController.class);
+
+    private static final String USER_ID_PARAM_NAME = "userId";
+    private static final String FROM_DATE_PARAM_NAME = "fromDate";
+    private static final String TO_DATE_PARAM_NAME = "toDate";
+    private static final String LIMIT_PARAM_NAME = "limit";
+    private static final int DEFAULT_LIMIT = 10;
+
+    /**
+     * Return JSON representation of a user's trip request history based on provided parameters.
+     * An authorized user (Auth0) and user id are required.
+     */
+    public static String getTripRequests(Request request, Response response, TypedPersistence<TripRequest> tripRequest) {
+
+        Auth0UserProfile requestingUser = Auth0Connection.getUserFromRequest(request);
+        if (requestingUser == null) {
+            logMessageAndHalt(request, HttpStatus.FORBIDDEN_403, "Authorization required.");
+        }
+
+        final String userId = HttpUtils.getParamFromRequest(request, USER_ID_PARAM_NAME, false);
+        OtpUser user = Persistence.otpUsers.getById(userId);
+        if (user == null) {
+            logMessageAndHalt(request, HttpStatus.FORBIDDEN_403, "Unknown user.");
```

Comment:
Should this be 404 instead?

Reply:
I've replace this section of code with isValidUser. Will update to 404 within that method, but under https://github.com/ibi-group/otp-middleware/pull/21